### ISSUE TITLE
assign in lodash/fp turns array into an object

### ISSUE THREAD

**Issue by AriaFallah** [lodash/lodash#2275]
Given the following code:

```js
import _ from 'lodash'
import fp from 'lodash/fp'

const array = []
console.log(fp.assign([1, 2, 3])(array))
console.log(_.partialRight(_.assign, [1, 2, 3])(array))
```

the result is

```js
{ '0': 1, '1': 2, '2': 3 }
[ 1, 2, 3 ]
```

Is this the expected behavior?

**Comment by jdalton** [comment]
This is by design for the fp flavor. I don't want it to get too clever on cloning the input.

**Comment by jdalton** [comment]
@jfmengels
Do you have an opinion on this?
I could check for `isArray` but not sure if that's a slippery slope in this case.

**Comment by jfmengels** [comment]
I don't see how this is by design in `fp`. I would like to know why the behavior is different in FP in this case and not for other functions that have arrays as args.

In this case, the docs stipulate that the arguments must be `objects`, not arrays (arrays are objects, but when arrays are expected, the docs say so explicitly), and I've never thought of using `assign` for working with arrays, so I don't think it should be handled. Sure, it's handled well in vanilla Lodash, but give it unexpected arguments, expected unexpected results.

The use cases for having `assign` for arrays as I see them:
- cloning an array (only with FP) (handle by `_.clone`)
- replacing the first few elements of an array (in a new array or not) like
```js
_.assign([1, 2, 3, 4, 5], [101, 102, 103])
// => [101, 102, 103, 4, 5]
```
which could be useful but seems like an edge case and you can use operations like `slice` for it.

tl;dr: I think `assign` should stick to objects only.

**Comment by jdalton** [comment]
This shakes out of FP because its methods are immutable so it has to create a fresh object. This also aligns with other FP libs behavior.

The non-FP flavor works with arrays or any other objects because it mutates and falls through the method as its output.
